model/edx.go

```diff
@@ -67,6 +67,45 @@ func (e *EdxLMS) GetOAuthConsumerSecret() string {
 	return e.OAuthConsumerSecret
 }
 
+func (e *EdxLMS) GetTeams() map[string]string {
+	return e.Teams
+}
+
+func (e *EdxLMS) CheckIfActiveLMS(currentTeamName string) bool {
+	// For every property of Teams in this learning management service (LMS) object (passed as e)
+	// The team's slug names are the values of each key, value pair
+	// Represent each team's slug name with teamSlug, _ as an empty placeholder for the key of this lms object
+	for _, teamSlug := range e.GetTeams() {
+		// If this team slug is equal to the team slug of the currently selected mattermost team in the web app
+		if teamSlug == currentTeamName {
+			return true
+		}
+	}
+	return false
+}
+
+func (e *EdxLMS) GetLearningGroupPrefixes() ([]*LearningGroupPrefix, *AppError) {
+	var personalChannelNames []*LearningGroupPrefix = make([]*LearningGroupPrefix, len(e.PersonalChannels.ChannelList))
+
+	// Loop over PersonalChannels.ChannelList in the active LMS configuration and, for every key,
+	// add a pointer to a LearningGroupPrefix struct containing this key (key is the prefix)
+	numberPrefixesStored := 0
+	for personalChannelName := range e.PersonalChannels.ChannelList {
+		personalChannelNames[numberPrefixesStored] = &LearningGroupPrefix{Prefix: personalChannelName}
+
+		numberPrefixesStored++
+	}
+
+	// Assert personalChannelNames as type 'array of pointers to LearningGroupPrefix structs'
+	var assertionSuccess bool
+	if personalChannelNames, assertionSuccess = interface{}(personalChannelNames).([]*LearningGroupPrefix); !assertionSuccess {
```

Comment:
Is this cast actually needed? Since `personalChannelNames` was declared to be of this type, shouldn't the assertion always succeed?

Reply:
Well you are manipulating the struct in a way that may or may not change the dynamic type of the variable. I am a little rusty on this terminology (dynamic and static types) but I wanted to be sure here.

Reply:
From stackoverflow:

> A dynamic type means that it can hold a reference to different types (e.g. string, int, ...) and that it can change at runtime, whereas a static type is checked at compile time and cannot change.

Reply:
NVM. Forgot that this is idiomatic check for nil in Golang.